Previous parallel make patches from user submission were not quite enough.

Include this bit as well to fix up bison being invoked multiple times.

X-Gentoo-Bug: 389799
X-Gentoo-Bug-URL: https://bugs.gentoo.org/389799
Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar acpica-unix-20110922.orig/compiler/Makefile acpica-unix-20110922/compiler/Makefile
--- acpica-unix-20110922.orig/compiler/Makefile	2011-11-07 11:14:51.200026000 +0000
+++ acpica-unix-20110922/compiler/Makefile	2011-11-07 11:20:03.490637284 +0000
@@ -274,14 +274,18 @@
 aslcompilerlex.c :   $(ASL_COMPILER)/aslcompiler.l
 	${LEX} ${LFLAGS} -PAslCompiler -o$@ $?
 
-aslcompiler.y.c aslcompiler.y.h aslcompiler.y.output : $(ASL_COMPILER)/aslcompiler.y
-	${YACC} ${YFLAGS} -pAslCompiler -oaslcompiler.y.c $?
+aslcompiler.y.c : $(ASL_COMPILER)/aslcompiler.y
+	${YACC} ${YFLAGS} -pAslCompiler -o aslcompiler.y.c $?
+aslcompiler.y.h : aslcompiler.y.c
+aslcompiler.y.output : aslcompiler.y.c
 
 dtparserlex.c :      $(ASL_COMPILER)/dtparser.l
 	${LEX} ${LFLAGS} -PDtParser -o$@ $?
 
-dtparser.y.c dtparser.y.h dtparser.y.output : $(ASL_COMPILER)/dtparser.y
-	${YACC} ${YFLAGS} -pDtParser -odtparser.y.c $?
+dtparser.y.c : $(ASL_COMPILER)/dtparser.y
+	${YACC} ${YFLAGS} -pDtParser -o dtparser.y.c $?
+dtparser.y.h : dtparser.y.c
+dtparser.y.output : dtparser.y.c
 
 # Rename headers produced by bison/yacc
 
@@ -294,14 +298,14 @@
 aslcompilerlex.o :   aslcompilerlex.c aslcompiler.y.h
 	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $<
 
-aslcompiler.y.o : aslcompiler.y.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+aslcompiler.y.o : aslcompiler.y.c aslcompiler.y.h
+	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $<
 
 dtparserlex.o :      dtparserlex.c dtparser.y.h
 	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $<
 
-dtparser.y.o :    dtparser.y.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+dtparser.y.o :    dtparser.y.c dtparser.y.h
+	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $<
 
 
 %.o : %.c $(HEADERS) $(ACPICA_HEADERS)
